SynthShop.Core/Services/Impl/AuthService.cs

```diff
@@ -26,18 +27,30 @@ public class AuthService : IAuthService
         private readonly IConfiguration _configuration;
         private readonly TokenValidationParameters _tokenValidationParameters;
         private readonly IAuthRepository _authRepository;
-
-        public AuthService(UserManager<User> userManager, IConfiguration configuration, TokenValidationParameters tokenValidationParameters, IAuthRepository authRepository)
+        private readonly ILogger _logger;
+        
+        public AuthService(UserManager<User> userManager, IConfiguration configuration, TokenValidationParameters tokenValidationParameters, IAuthRepository authRepository, ILogger logger)
         {
             _userManager = userManager;
             _configuration = configuration;
             _tokenValidationParameters = tokenValidationParameters;
             _authRepository = authRepository;
+            _logger = logger;
         }
         public async Task<IdentityResult> RegisterUserAsync(User user, string password)
         {
 
+
+
             var result = await _userManager.CreateAsync(user, password);
+            if (result.Succeeded)
+            {
+                _logger.Information("User {Email} registered successfully", user.Email);
+            }
+            else
+            {
+                _logger.Warning("User registration failed for {Email}. Errors: {Errors}", user.Email, result.Errors);
```

Comment:
Need to use the Error level, and add @ (like: Errors: {@Errors}) logging result.Errors because the type of result.Errors contain IEnumerable so we need to destruct it.